core/src/test/java/BookServiceTest.java

```diff
@@ -0,0 +1,200 @@
+import com.ruslan.database.DAO.BookRepository;
+import com.ruslan.database.DAO.OrderRepository;
+import com.ruslan.database.DAO.RequestRepository;
+import com.ruslan.entity.book.Book;
+import com.ruslan.entity.book.BookStatus;
+import com.ruslan.entity.order.Order;
+import com.ruslan.entity.order.OrderStatus;
+import com.ruslan.entity.request.Request;
+import com.ruslan.json.JsonReader;
+import com.ruslan.services.BookService;
+import com.ruslan.services.OrderService;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private RequestRepository requestRepository;
+    @Mock
+    private OrderRepository orderRepository;
+    @InjectMocks
+    private BookService bookService;
+    @InjectMocks
+    private OrderService orderService;
+    private final JsonReader jsonReader = JsonReader.getInstance();
+    public String pathBookSJSON = "src\\test\\resources\\BooksForTest.json";
+    public String pathRequestSJSON = "src\\test\\resources\\RequestsForTest.json";
+    public String pathOrderJSON = "src\\test\\resources\\OrdersForTest.json";
+
+    List<Book> bookList;
+    List<Request> requestList;
+    List<Order> orderList;
+
+    @BeforeEach
+    void setUp() throws NoSuchFieldException, IllegalAccessException {
+        bookList = jsonReader.readEntities(Book.class, pathBookSJSON);
+
+        requestList = jsonReader.readEntities(Request.class, pathRequestSJSON);
+
+        orderList = jsonReader.readEntities(Order.class, pathOrderJSON);
+
+        initService();
+    }
+
+
+    @Test
+    void addBookToStockAndCancelRequests() {
```

Comment:
Good practice is to add test outcome to method name. In that way your team will understand from beginning what test checking. For example: addBookToStockAndCancelRequests_true, addBookToStockAndCancelRequests_returnList,
addBookToStockAndCancelRequests_throwsException, addBookToStockAndCancelRequests_onlyOnce, etc.

Reply:
Changed names